pybluedot/cli.py

```diff
@@ -1,7 +1,46 @@
-"""
-cli.py constructs a cli interface of commands, options, and subcommands in argparse, given data structures from commands.py
-"""
 import argparse
-import json
-from pybluedot.commands import cmd, subcmd 
+from pybluedot.commands import cmd_arg
+from pybluedot.commands import sub_cmd
 
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+
+def cmd_to_args(cmd_arg):
+    """creates argparse arguments from commands.cmd dict.
+    return argparse object.
+    """
+    all_args = argparse.ArgumentParser()
+    for arg in cmd_arg:
+        for i in sorted(arg.keys()):
+            # TODO(gamefiend)
+            pass
+    return all_args
+
+
+def cmd_to_subs(sub_cmd):
+    """creates subcommand arguments from commands.subcmd dict.
+        returns argparse object.
+    """
+    all_subs = argparse.ArgumentParser()
+    for sub in sub_cmd:
+        for i in sorted(sub):
+            # TODO(gamefiend)
+            pass
+    return all_subs
+
+
+if __name__ == "__main__":
```

Comment:
See https://github.com/commitmas/pybluedot/blob/master/setup.cfg#L24 and study up on the setuptools configuration. In short, none of the Python files we develop will be called directly - that role will be fulfilled by a console script ("binary") generated by setuptools. What you should do is create a "main()" function here so that the current setuptools configuration knows what to grab.

Then, running this is simple:

```
sudo python setup.py install
bdot patents
```